yt_dlp/extractor/callin.py

```diff
@@ -0,0 +1,135 @@
+        show_id = traverse_obj(episode, ('show', 'id'))
+
+        # get the show metadata for some supplementary info
+        sj = None
+        try:
+            app_slug = self._html_search_regex(
+                '<script\\s+src=["\']/_next/static/([a-zA-Z0-9_]+)/_',
+                webpage, 'app slug', fatal=False)
+            if app_slug is None:
+                # this sometimes works, but sometimes seems to lag changes in the actual urls used.
+                app_slug = j['buildId']
+            show_slug = episode['show']['linkObj']['resourceUrl'].rsplit('/', 1)[1]
+            show_json_url = "https://www.callin.com/_next/data/" + app_slug + "/show/" + show_slug + ".json"
+
+            sj = self._download_json(show_json_url, id)
+        except (KeyError, IndexError, ExtractorError):
+            pass
+
+        host = traverse_obj(sj, ('pageProps', 'show', 'hosts', 0))
+        if host is None:
+            # guess that it's the first speaker listed
+            host = traverse_obj(episode, ('speakers', 0))
+
+        host_url = traverse_obj(sj, ('pageProps', 'show', 'url'))
+
+        host_nick = traverse_obj(host, ('linkObj', 'resourceUrl'))
+        host_nick = host_nick.rsplit('/', 1)[1] if (host_nick and '/' in host_nick) else None
+
+        categories = list(filter(None, [
+            c.get('name') for c in
+            traverse_obj(episode, ('show', 'categorizations')) or []
+        ]))
+
+        cast = list(filter(None, [
+            self.try_get_user_name(u) for u in
+            ((episode.get('speakers') or []) + (episode.get('callerTags') or []))
+        ]))
+
+        episode_number = None
+        if sj:
+            episode_list = traverse_obj(sj, ('pageProps', 'show', 'episodes')) or []
+            for (i, e) in enumerate(episode_list):
+                if e.get('id') == id:
+                    episode_number = len(episode_list) - i
+                    break
+
+        return {
+            'id': id,
+            'display_id': display_id,
+            'title': title,
+            'formats': formats,
+            'thumbnail': traverse_obj(episode, ('show', 'photo')),
+            'description': episode.get('description'),
+            'uploader': self.try_get_user_name(host) if host else None,
+            'timestamp': episode.get('publishedAt'),
+            'uploader_id': host_nick,
+            'uploader_url': host_url,
+            'channel': show,
+            'channel_id': show_id,
+            'channel_url': traverse_obj(episode, ('show', 'linkObj', 'resourceUrl')),
+            'duration': float_or_none(episode.get('runtime')),
+            'view_count': int_or_none(episode.get('plays')),
+            'categories': categories if categories else None,
+            'cast': cast if cast else None,
+            'series': show,
+            'series_id': show_id,
+            'episode': title,
+            'episode_number': episode_number,
+            'episode_id': id
```

Comment:
I think a few more of these can be inlined

Reply:
Done (in 0854618e72ab8b35847c2acefb2f3f6405b780b9), unless I've missed any.